website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,764 @@
+            if (result) {
+                confirm();
+            }
+        }
+    );
+};
+Draft.prototype.beforeRegister = function(data) {
+    var self = this;
+
+    $osf.block();
+
+    return $.getJSON(self.urls.before_register).then(function(response) {
+        if (response.errors && response.errors.length) {
+            self.preRegisterErrors(response, self.preRegisterWarnings);
+        } else if (response.prompts && response.prompts.length) {
+            self.preRegisterPrompts(response, self.register.bind(self, data));
+        } else {
+            self.register(data);
+        }
+    }).always($osf.unblock);
+};
+Draft.prototype.onRegisterFail = bootbox.alert.bind(null, {
+    title: 'Registration failed',
+    message: language.registerFail
+});
+Draft.prototype.register = function(data) {
+    var self = this;
+
+    $osf.block();
+
+    $osf.postJSON(self.urls.register, data)
+        .done(function(response) {
+            if (response.status === 'initiated') {
+                window.location.assign(response.urls.registrations);
+            } else if (response.status === 'error') {
+                self.onRegisterFail();
+            }
+        }).always($osf.unblock).fail(self.onRegisterFail);
+};
+
+
+/**
+ * @class RegistrationEditor
+ *
+ * @param {Object} urls
+ * @param {String} urls.update: endpoint to update a draft instance
+ * @param {String} editorID: id of editor DOM node
+ * @property {ko.observable[Boolean]} readonly
+ * @property {ko.observable[Draft]} draft
+ * @property {ko.observable[Question]} currentQuestion
+ * @property {Object} extensions: mapping of extenstion names to their view models
+ **/
+var RegistrationEditor = function(urls, editorId) {
+
+    var self = this;
+
+    self.urls = urls;
+
+    self.readonly = ko.observable(false);
+
+    self.draft = ko.observable();
+
+    self.currentQuestion = ko.observable();
+    // When the currentQuestion changes, save when it's rate-limited value changes
+    self.currentQuestion.subscribe(function(question) {
+         question.delayedValue.subscribe(self.save.bind(self));
+    });
+    self.showValidation = ko.observable(false);
+
+    self.currentPages = ko.computed(function() {
+        var draft = self.draft();
+        if (!draft) {
+            return [];
+        }
+        var schema = draft.schema();
+        if (!schema) {
+            return [];
+        }
+        return schema.pages;
+    });
+
+    self.lastSaveTime = ko.computed(function() {
+        if (!self.draft()) {
+            return null;
+        }
+        return self.draft().updated;
+    });
+    self.lastSaved = ko.computed(function() {
+        var t = self.lastSaveTime();
+        if (t) {
+            return t.toGMTString();
+        } else {
+            return 'never';
+        }
+    });
+
+    self.canRegister = ko.computed(function() {
+        var draft = self.draft();
+        return draft && draft.isApproved;
+    });
+
+    self.iterObject = $osf.iterObject;
```

Comment:
Eric has mentioned to me that the function was removed, but this template was not updated.